### PR TITLE
Replace custom RMSNorm layers with keras.layers.RMSNormalization

### PR DESCRIPTION
**Pull request by Ahmed0830**
## Description of the change
Nine model families (llama, mistral, mixtral, qwen, qwen_moe, phi3, gpt_oss, qwen3, qwen3_moe) each carried their own hand copied RMSNorm layer, byte-for-byte identical math (float32-upcast mean/rsqrt/scale) reimplemented per family. Swapped all of them for Keras 3's built-in `keras.layers.RMSNormalization`, which does the same computation and gets the fused-kernel path on the torch backend for free.

Left untouched, out of scope for this pass: the Gemma family (zeros-init `scale` + `(1 + scale)` convention has no equivalent in the native layer), and a handful of files with genuinely different behavior (gemma3n, edrec, t5, gemma4's VNorm/FrozenNorm).

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: keras_hub/src/models/qwen3/qwen3_attention.py [hunk 1]
```diff
@@ -84,13 +83,12 @@ def build(self, inputs_shape):
...
 
-        self._query_dense_layer_norm = Qwen3LayerNorm(
+        self._query_dense_layer_norm = keras.layers.RMSNormalization(
             epsilon=self.layer_norm_epsilon,
             dtype=self.dtype_policy,
-            head_dim=self.head_dim,
             name="query_dense_layernorm",
         )
-        self._query_dense_layer_norm.build(inputs_shape)
+        self._query_dense_layer_norm.build((None, None, self.head_dim))
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/models/qwen3/qwen3_attention.py]: fixed, thanks

File: keras_hub/src/models/qwen3/qwen3_attention.py [hunk 2]
```diff
@@ -105,13 +103,12 @@ def build(self, inputs_shape):
...
 
-        self._key_dense_layer_norm = Qwen3LayerNorm(
+        self._key_dense_layer_norm = keras.layers.RMSNormalization(
             epsilon=self.layer_norm_epsilon,
             dtype=self.dtype_policy,
-            head_dim=self.head_dim,
             name="key_dense_layernorm",
         )
-        self._key_dense_layer_norm.build(inputs_shape)
+        self._key_dense_layer_norm.build((None, None, self.head_dim))
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/models/qwen3/qwen3_attention.py]: fixed, thanks

File: keras_hub/src/models/qwen3_moe/qwen3_moe_attention.py [hunk 3]
```diff
@@ -85,13 +84,12 @@ def build(self, inputs_shape):
...
 
-        self._query_dense_layer_norm = Qwen3MoeLayerNorm(
+        self._query_dense_layer_norm = keras.layers.RMSNormalization(
             epsilon=self.layer_norm_epsilon,
             dtype=self.dtype_policy,
-            head_dim=self.head_dim,
             name="query_dense_layernorm",
         )
-        self._query_dense_layer_norm.build(inputs_shape)
+        self._query_dense_layer_norm.build((None, None, self.head_dim))
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/models/qwen3_moe/qwen3_moe_attention.py]: fixed, thanks

File: keras_hub/src/models/qwen3_moe/qwen3_moe_attention.py [hunk 4]
```diff
@@ -106,13 +104,12 @@ def build(self, inputs_shape):
...
 
-        self._key_dense_layer_norm = Qwen3MoeLayerNorm(
+        self._key_dense_layer_norm = keras.layers.RMSNormalization(
             epsilon=self.layer_norm_epsilon,
             dtype=self.dtype_policy,
-            head_dim=self.head_dim,
             name="key_dense_layernorm",
         )
-        self._key_dense_layer_norm.build(inputs_shape)
+        self._key_dense_layer_norm.build((None, None, self.head_dim))
```
[the remark this answers is not in the document]
Comment (Ahmed0830) [reply keras_hub/src/models/qwen3_moe/qwen3_moe_attention.py]: fixed, thanks

**Comment by Ahmed0830** [comment]
I’ve extended this PR to cover `Flux` as well. Flux had its own separate `RMSNormalization` helper (keras_hub/src/layers/modeling/rms_normalization.py), which was only used by `QKNorm`. I swapped that out for `keras.layers.RMSNormalization()` directly, so there’s no need for a wrapper.

`QKNorm.build()` already receives a shape whose last axis is head_dim (from SelfAttention.build() / SingleStreamBlock.build()), so the built-in’s default `axis=-1` gives us the same weight shape that the old input_dim constructor arg did.

This also removes the `keras_hub.layers.RMSNormalization` public export as no other model used this, so I wanted to call that out explicitly since it’s an API removal, not just an internal refactor.

**Comment by laxmareddyp** [comment]
Hi @Ahmed0830 ,

Thanks so much for putting this PR together! Conceptually this makes sense, but I'm going to hold off on merging it for now.A few reasons on my end: 

Risk vs. Reward:  We’d be touching 32 files across 9 model families, all to replace custom RMS layers that are already simple, well-tested, and working fine. Switching to keras.layers.RMSNormalization gives us very little gain, but if anything breaks, the impact is huge.

It is a breaking API change since keras_hub.layers.RMSNormalization is exported publicly. We can't just delete it in one PR without a deprecation period first.

Verification : For a change this broad, we'd need concrete numerical parity evidence Ex: a Colab showing identical outputs before/after for at least a few model families with loaded presets

If you'd like to pursue this, I'd suggest starting much smaller — pick one model family  provide numerical verification with a loaded preset, and we can work on other family models ,  But honestly, this isn't a high-priority cleanup right now.

Thanks.

**Comment by Ahmed0830** [comment]
> Hi @Ahmed0830 ,
> 
> Thanks so much for putting this PR together! Conceptually this makes sense, but I'm going to hold off on merging it for now.A few reasons on my end:
> 
> Risk vs. Reward: We’d be touching 32 files across 9 model families, all to replace custom RMS layers that are already simple, well-tested, and working fine. Switching to keras.layers.RMSNormalization gives us very little gain, but if anything breaks, the impact is huge.
> 
> It is a breaking API change since keras_hub.layers.RMSNormalization is exported publicly. We can't just delete it in one PR without a deprecation period first.
> 
> Verification : For a change this broad, we'd need concrete numerical parity evidence Ex: a Colab showing identical outputs before/after for at least a few model families with loaded presets
> 
> If you'd like to pursue this, I'd suggest starting much smaller — pick one model family provide numerical verification with a loaded preset, and we can work on other family models , But honestly, this isn't a high-priority cleanup right now.
> 
> Thanks.

Totally fair, thanks for the detailed breakdown @laxmareddyp. Agreed this is low-priority cleanup and not worth multiple review cycles right now.

For context on why I raised it: Keras has built-in RMSNormalization support now, so it should be preferred going forward, but when porting models with reference to other implementations, it's easy to end up copying over a custom RMSNorm layer that Keras already natively supports, without realizing it. This PR was really just flagging that.

I'll convert this to a draft and park it for now, happy to revisit later if it becomes worth prioritizing.
